#!/usr/bin/env python
PACKAGE = "syropod_highlevel_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("step_frequency", double_t, 0, "Frequency of entire step cycle of one leg (Hz)", 1.0)
gen.add("swing_height", double_t, 0, "Vertical displacement of swing trajectory above default position (metres)", 1.0)
gen.add("swing_width", double_t, 0, "Lateral displacement of swing trajectory out from default position (metres)", 1.0) 
gen.add("step_depth", double_t, 0, "Depth below walk plane step will target to find contact (metres)", 1.0)
gen.add("stance_span_modifier", double_t, 0, "Modifier for inc/dec stance span (y axis)", 1.0)
gen.add("virtual_mass", double_t, 0, "Mass of virtual system of impedance controller", 1.0)
gen.add("virtual_stiffness", double_t, 0, "Stiffness of virtual system of impedance controller", 1.0)
gen.add("virtual_damping_ratio", double_t, 0, "Damping ratio of virtual system of impedance controller", 1.0)
gen.add("force_gain", double_t, 0, "Multiplier of input force for virtual system of impedance controller", 1.0)

exit(gen.generate(PACKAGE, "syropod_highlevel_controller", "Dynamic"))
